src/App.js

```diff
@@ -1,19 +1,44 @@
 import 'bootstrap/dist/css/bootstrap.min.css';
 import './App.css';
+import {Component} from 'react'
 import Header from './components/Header'
 import Main from './components/Main'
 import Footer from './components/Footer'
+import SelectedBeast from './components/SelectedBeast'
 import beastObjects from './data.json'
 
 
-function App() {
-  return (
-    <>
-    <Header title='The Beastie Boys 🦏'/>
-    <Main message='Beast selection' beastObjects={beastObjects}/>
-    <Footer title='Author: Jacob Gregor'/>
-    </>
-  );
+class App extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      showBeast: true,
+      beast: beastObjects[0]
+    }
+  };
+    
+    modalOnClick = (beast) => {this.setState ({ 
+      showBeast: true,
+      beast: beast
+      
+    }); console.log('test')}
+ 
+
+    modalOnClose = () => this.setState ({ showBeast: false })
+  
+
+
+  render() {
+    return (
+      <>
+      <Header title='The Beastie Boys 🦏'/>
+      <Main modalClick={this.modalOnClick} message='Beast selection' beastObjects={beastObjects}/>
+      <SelectedBeast beast={this.state.beast} show={this.state.showBeast} modalClose={this.modalOnClose}/>
```

Comment:
Great work here!

Just a thought for when you start to see a bunch of props passed to a component: it helps to indent each one once you reach 3 or more...not a necessity but it is common to see...not a biggie here, just a thought :)

```
 <Main 
   modalClick={this.modalOnClick} 
   message='Beast selection' 
   beastObjects={beastObjects}
 />
```